linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -1029,12 +1029,30 @@ protected TypeReference GetOriginalType (TypeReference type)
 				if (mod != null)
 					MarkModifierType (mod);
 
-				type = ((TypeSpecification) type).ElementType;
+				var fnptr = type as FunctionPointerType;
+				if (fnptr != null) {
+					MarkParameters (fnptr);
+					type = fnptr.ReturnType;
```

Comment:
I think it would be better to call MarkType (fnptr.ReturnType) here and just return. Function's return type is not the original type of the function.